app/src/main/java/com/google/samples/apps/sunflower/adapters/PlantAdapter.kt

```diff
@@ -47,29 +44,25 @@ class PlantAdapter : ListAdapter<Plant, PlantAdapter.ViewHolder>(PlantDiffCallba
     }
 
     override fun onBindViewHolder(holder: ViewHolder, position: Int) {
-        getItem(position).apply {
-            holder.itemView.tag = this
-            with(holder.binding) {
-                setVariable(BR.vm, ItemViewModel(this@apply, onClickListener))
-                executePendingBindings()
-            }
+        val plant = getItem(position)
+        holder.apply {
+            bind(onClickListener, plant)
+            itemView.tag = plant
         }
     }
 
-    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int) = ViewHolder(
-        DataBindingUtil.inflate(
-            LayoutInflater.from(parent.context),
-            R.layout.list_item_plant, parent, false
-        )
-    )
-
-    class ViewHolder(val binding: ViewDataBinding) : RecyclerView.ViewHolder(binding.root)
-
-    class ItemViewModel(plant: Plant, val clickHandler: View.OnClickListener) :
-        ViewModel() {  // Actually no need, however, for unified namespace with other [ViewModels]s I consider it.
-
-        val content = ObservableField<String>(plant.name)
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
+        return ViewHolder(ListItemPlantBinding.inflate(
+                LayoutInflater.from(parent.context), parent, false))
+    }
 
-        val imageUrl = ObservableField<String>(plant.imageUrl)
+    class ViewHolder(private val binding: ViewDataBinding) : RecyclerView.ViewHolder(binding.root) {
```

Comment:
How about writing like this?
It uses ListItemPlantBinding instead of ViewDataBinding so that variables can be accessed directly instead of using setVariable


```kt
    class ViewHolder(private val binding: ListItemPlantBinding) : RecyclerView.ViewHolder(binding.root) {
        fun bind(listener: View.OnClickListener, item: Plant) {
            binding.apply {
                clickListener = listener
                plant = item
                executePendingBindings()
            }
        }
    }
```

Reply:
Oh, neat! I applied your suggestion.  PTAL